crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,818 @@
+    let (opening_paren, closing_paren) = kind.surrounding_parens(locator.contents());
+    let last_item_index = elements.len().saturating_sub(1);
+    let mut result = String::from(opening_paren);
+
+    let mut element_pairs = elements.iter().zip(elts).collect_vec();
+    element_pairs.sort_by(|(elem1, _), (elem2, _)| cmp_fn(elem1, elem2));
+    // We grab the original source-code ranges using `locator.slice()`
+    // rather than using the expression generator, as this approach allows
+    // us to easily preserve stylistic choices in the original source code
+    // such as whether double or single quotes were used.
+    for (i, (_, elt)) in element_pairs.iter().enumerate() {
+        result.push_str(locator.slice(elt));
+        if i < last_item_index {
+            result.push_str(", ");
+        }
+    }
+
+    result.push_str(closing_paren);
+    result
+}
+
+/// Create a string representing a fixed-up single-line
+/// definition of `__all__` or `__slots__` (etc.),
+/// that can be inserted into the
+/// source code as a `range_replacement` autofix.
+pub(super) fn sort_single_line_elements_dict<F>(
+    key_elts: &[ast::Expr],
+    elements: &[&str],
+    value_elts: &[ast::Expr],
+    locator: &Locator,
+    mut cmp_fn: F,
+) -> String
+where
+    F: FnMut(&str, &str) -> Ordering,
+{
+    assert!(key_elts.len() == elements.len() && elements.len() == value_elts.len());
+    let last_item_index = elements.len().saturating_sub(1);
+    let mut result = String::from('{');
+
+    let mut element_trios = izip!(elements, key_elts, value_elts).collect_vec();
+    element_trios.sort_by(|(elem1, _, _), (elem2, _, _)| cmp_fn(elem1, elem2));
+    // We grab the original source-code ranges using `locator.slice()`
+    // rather than using the expression generator, as this approach allows
+    // us to easily preserve stylistic choices in the original source code
+    // such as whether double or single quotes were used.
+    for (i, (_, key, value)) in element_trios.iter().enumerate() {
+        result.push_str(locator.slice(key));
+        result.push_str(": ");
+        result.push_str(locator.slice(value));
+        if i < last_item_index {
+            result.push_str(", ");
+        }
+    }
+
+    result.push('}');
+    result
+}
+
+/// An enumeration of the possible conclusions we could come to
+/// regarding the ordering of the elements in a display of string literals:
+///
+/// 1. It's a display of string literals that is already sorted
+/// 2. It's an unsorted display of string literals,
+///    but we wouldn't be able to autofix it
+/// 3. It's an unsorted display of string literals,
+///    and it's possible we could generate a fix for it
+/// 4. The display contains one or more items that are not string
+///    literals.
+#[derive(Debug, is_macro::Is)]
+pub(super) enum SortClassification<'a> {
+    Sorted,
+    UnsortedButUnfixable,
+    UnsortedAndMaybeFixable { items: Vec<&'a str> },
+    NotAListOfStringLiterals,
+}
```

Comment:
Nit: I assume they are in order. Moving the documentation to the variants removes the need for numbers and ensures the documentation stays up to date even when re-ordering variants
```suggestion

/// 3. It's an unsorted display of string literals,
///    and it's possible we could generate a fix for it
/// 4. The display contains one or more items that are not string
///    literals.
#[derive(Debug, is_macro::Is)]
pub(super) enum SortClassification<'a> {
    /// A display of string literals that is already sorted
    Sorted,
    /// An unsorted display of string literals, but ruff isn't be able to autofix it.
    UnsortedButUnfixable,
    UnsortedAndMaybeFixable { items: Vec<&'a str> },
    NotAListOfStringLiterals,
}
```